src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubScaleMonitor.cs

```diff
@@ -0,0 +1,395 @@
+            ILogger logger,
+            CloudBlobContainer blobContainer = null)
+        {
+            _functionId = functionId;
+            _eventHubName = eventHubName;
+            _consumerGroup = consumerGroup;
+            _logger = logger;
+            _scaleMonitorDescriptor = new ScaleMonitorDescriptor($"{_functionId}-EventHubTrigger-{_eventHubName}".ToLower());
+            _nextPartitionLogTime = DateTime.UtcNow;
+            _nextPartitionWarningTime = DateTime.UtcNow;
+
+            _connectionStringBuilder = new EventHubsConnectionStringBuilder(connectionString);
+            if (!String.IsNullOrEmpty(_eventHubName))
+            {
+                _connectionStringBuilder.EntityPath = _eventHubName;
+            }
+            _client = EventHubClient.CreateFromConnectionString(_connectionStringBuilder.ToString());
+
+            _blobContainer = blobContainer;
+            if (_blobContainer == null)
+            {
+                CloudStorageAccount cloudStorageAccount = CloudStorageAccount.Parse(storageConnectionString);
+                CloudBlobClient blobClient = cloudStorageAccount.CreateCloudBlobClient();
+                _blobContainer = blobClient.GetContainerReference(EventHubContainerName);
+            }
+        }
+
+        public ScaleMonitorDescriptor Descriptor
+        {
+            get
+            {
+                return _scaleMonitorDescriptor;
+            }
+        }
+
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await GetMetricsAsync();
+        }
+
+        public async Task<EventHubTriggerMetrics> GetMetricsAsync()
+        {
+            EventHubTriggerMetrics metrics = new EventHubTriggerMetrics();
+            EventHubRuntimeInformation runtimeInfo = null;
+
+            try
+            {
+                runtimeInfo = await _client.GetRuntimeInformationAsync();
+            }
+            catch (NotSupportedException e)
+            {
+                _logger.LogWarning($"Event Hub Trigger does not support Notification Hubs. Error: {e.Message}");
+                return metrics;
+            }
+            catch (MessagingEntityNotFoundException)
+            {
+                _logger.LogWarning($"Event Hub '{_eventHubName}' was not found.");
+                return metrics;
+            }
+            catch (TimeoutException e)
+            {
+                _logger.LogWarning($"Encountered a timeout while checking Event Hub '{_eventHubName}'. Error: {e.Message}");
+                return metrics;
+            }
+            catch (Exception e)
+            {
+                _logger.LogWarning($"Encountered an exception while checking Event Hub '{_eventHubName}'. Error: {e.Message}");
+                return metrics;
+            }
+
+            // Get the PartitionRuntimeInformation for all partitions
+            var tasks = new Task<EventHubPartitionRuntimeInformation>[runtimeInfo.PartitionCount];
+

```

Comment:
Can you add a log statement here to the effect "Querying partition information for {partitionCount} partitions"?